src/blop/agent.py

```diff
@@ -293,101 +435,370 @@ def ask(self, acqf="qei", n=1, route=True, sequential=True, upsample=1, **acqf_k
+        self._acqf_kwargs = acqf_kwargs
+        self._detector_names = list(detector_names)
+
+    @property
+    def detector_names(self):
+        return [str(name) for name in self._detector_names]
+
+    @detector_names.setter
+    def detector_names(self, names):
+        self._detector_names = list(names)
+
+    @property
+    def acquisition_function(self):
+        return str(self._acqf_string)
+
+    @acquisition_function.setter
+    def acquisition_function(self, acqf_string):
+        self._acqf_string = str(acqf_string)
+
+    @property
+    def route(self):
+        return bool(self._route)
+
+    @route.setter
+    def route(self, route):
+        self._route = route
+
+    @property
+    def sequential(self):
+        return bool(self._sequential)
+
+    @sequential.setter
+    def sequential(self, sequential):
+        self._sequential = sequential
+
+    @property
+    def upsample(self):
+        return int(self._upsample)
+
+    @upsample.setter
+    def upsample(self, upsample):
+        self._upsample = int(upsample)
+
+    @property
+    def acqf_kwargs(self):
+        return {str(k): str(v) for k, v in self._acqf_kwargs.items()}
+
+    def update_acqf_kwargs(self, **kwargs):
+        self._acqf_kwargs.update(kwargs)
+
+    @property
+    def detector_names(self):
+        return [str(name) for name in self._detector_names]
+
+    @detector_names.setter
+    def detector_names(self, names):
+        self._detector_names = list(names)
+
+    def server_registrations(self):
+        """This is how we make these avaialble to the REST API."""
+        self._register_method("Update Acquistion Function Kwargs", self.update_acqf_kwargs)
+        self._register_property("Acquisition Function", self.acquisition_function, self.acquisition_function)
+        self._register_property("Route Points", self.route, self.route)
+        self._register_property("Sequential Points", self.sequential, self.sequential)
+        self._register_property("Upsample Points", self.upsample, self.upsample)
+        return super().server_registrations()
+
+    def ask(self, batch_size) -> Tuple[Sequence[Dict[str, ArrayLike]], Sequence[ArrayLike]]:
+        default_result = super().ask(
+            n=batch_size,
+            acqf=self._acqf_string,
+            route=self._route,
+            sequential=self._sequential,
+            upsample=self._upsample,
+            **self._acqf_kwargs,
+        )
+
+        """res = {
+            "points": {dof.name: list(points[..., i]) for i, dof in enumerate(active_dofs(read_only=False))},
+            "acqf_name": acqf_config["name"],
+            "acqf_obj": list(np.atleast_1d(acqf_obj.numpy())),
+            "acqf_kwargs": acqf_kwargs,
+            "duration_ms": duration,
+            "sequential": sequential,
+            "upsample": upsample,
+            "read_only_values": read_only_values,
+            # "posterior": p,
+        }
+        """
+
+        points: Dict = default_result.pop("points")
```

Comment:
Add stricter typing here, if possible. It looks like we make some assumptions about `points` below other than the fact that its a dictionary.